libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -751,16 +825,40 @@ void AP_UAVCAN::handle_actuator_status(AP_UAVCAN* ap_uavcan, uint8_t node_id, co
  */
 void AP_UAVCAN::handle_ESC_status(AP_UAVCAN* ap_uavcan, uint8_t node_id, const ESCStatusCb &cb)
 {
+    const uint8_t esc_index = cb.msg->esc_index;
+    
     // log as CESC message
     AP::logger().Write_ESCStatus(AP_HAL::micros64(),
-                                 cb.msg->esc_index,
+                                 esc_index,
                                  cb.msg->error_count,
                                  cb.msg->voltage,
                                  cb.msg->current,
                                  cb.msg->temperature - C_TO_KELVIN,
                                  cb.msg->rpm,
                                  cb.msg->power_rating_pct);
 
+    WITH_SEMAPHORE(_telem_sem);
+
+    if (!is_esc_data_index_valid(esc_index)) {
+        return;
+    }
+
+    esc_data &esc = _escs_data[esc_index];
+    esc.available = true;
+    esc.temp = (cb.msg->temperature - C_TO_KELVIN);
+    esc.voltage = cb.msg->voltage*100;
+    esc.current = cb.msg->current*100;
+    esc.rpm = cb.msg->rpm;
+    esc.count++;
+
+}
+
+bool AP_UAVCAN::is_esc_data_index_valid(const uint8_t index) {
+    if (index > UAVCAN_SRV_NUMBER) {
+        printf("UAVCAN: invalid esc index: %d. max index allowed: %d\n\r", index, UAVCAN_SRV_NUMBER);
```

Comment:
Please comment this out.

Reply:
Will do. The consequence is that the user will not be notified in case of an out of bound index.

Reply:
Understood.  Lesser of two evils.

@bugobliterator has offered to test this PR for us.

Reply:
Great news. Thanks.